ballista/rust/scheduler/src/scheduler_server/grpc.rs

```diff
@@ -406,7 +409,10 @@ impl<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan> SchedulerGrpc
 
             debug!("Received plan for execution: {:?}", plan);
 
-            let job_id = self.state.task_manager.generate_job_id();
+            let job_id = match optional_job_id {
+                Some(OptionalJobId::JobId(job_id)) => job_id,
```

Comment:
I think we need to check for a conflict here

Reply:
I tweaked the `get_job_status` signature a bit, so now we can validate if this job_id exists